addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightDiscover.java

```diff
@@ -77,61 +86,59 @@ public MilightDiscover(DiscoverResult discoverResult, int resendTimeoutInMillis,
     }
 
     public void dispose() {
+        stopResend();
         willbeclosed = true;
         datagramSocket.close();
         try {
             join(500);
```

Comment:
> Should I guard the join with a Thread.currentThread() != this?

Auch, if you state it like that it clarifies it all. Of course you can call `this.join()` as long as you don't call  `Thread.currentThread().join()` I'll go to bed early tonight :-S